File: apps/mobile/lib/screens/account_screen.dart
// account_screen.dart — todo lo del usuario: perfil, verificación de correo,
// tema, privacidad, API key, documentación (abre la web), borrado de
// historial, cierre de sesión y eliminación de cuenta.
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import 'package:url_launcher/url_launcher.dart';

import '../api.dart';
import '../state.dart';
import '../theme.dart';
import '../widgets/common.dart';

class AccountScreen extends StatelessWidget {
  const AccountScreen({super.key});

  void _snack(BuildContext context, String message) {
    ScaffoldMessenger.of(context).showSnackBar(SnackBar(content: Text(message)));
  }

  Future<void> _editName(BuildContext context) async {
    final api = context.read<ApiClient>();
    final auth = context.read<AuthState>();
    final user = auth.user ?? {};
    final current = [user['first_name'], user['last_name']]
        .whereType<String>()
        .where((s) => s.isNotEmpty)
        .join(' ');
    final value = await showPromptDialog(
      context,
      title: 'Tu nombre',
      message: 'Nombre y apellido separados por un espacio.',
      placeholder: 'Nombre Apellido',
      initialValue: current,
      confirmLabel: 'Guardar',
    );
    final trimmed = value?.trim() ?? '';
    if (trimmed.isEmpty) return;
    final parts = trimmed.split(RegExp(r'\s+'));
    final firstName = parts.first;
    final lastName = parts.length > 1
        ? parts.sublist(1).join(' ')
        : ((user['last_name'] as String?) ?? '');
    try {
      await api.patch('/api/account/profile',
          body: {'first_name': firstName, 'last_name': lastName});
      await auth.refreshMe();
    } on ApiException catch (err) {
      if (context.mounted) _snack(context, err.message);
    }
  }

  Future<void> _toggleSetting(BuildContext context, String key, bool value) async {
    final api = context.read<ApiClient>();
    final auth = context.read<AuthState>();
    try {
      await api.patch('/api/account/settings', body: {key: value});
      await auth.refreshMe();
    } on ApiException catch (err) {
      if (context.mounted) _snack(context, err.message);
    }
  }

  Future<void> _resendVerification(BuildContext context) async {
    final api = context.read<ApiClient>();
    try {
      final res = await api.post('/api/auth/resend-verification');
      if (context.mounted) {
        _snack(context, (res is Map ? res['message'] : null)?.toString() ?? 'Correo enviado');
      }
    } on ApiException catch (err) {
      if (context.mounted) _snack(context, err.message);
    }
  }

  Future<void> _regenerateKey(BuildContext context) async {
    final api = context.read<ApiClient>();
    final auth = context.read<AuthState>();
    final ok = await showConfirmDialog(
      context,
      title: 'Regenerar API key',
      message: 'Tu key actual dejará de funcionar en todos los dispositivos '
          '(esta app adoptará la nueva automáticamente).',
      confirmLabel: 'Regenerar',
      danger: true,
    );
    if (!ok) return;
    try {
      final res = await api.post('/api/auth/api-key/regenerate');
      final newKey = (res is Map ? res['newApiKey'] : null);
      if (newKey is String && newKey.isNotEmpty) {
        await auth.adoptApiKey(newKey);
      }
      if (context.mounted) {
        _snack(context, 'API key regenerada. Esta app ya usa la nueva.');
      }
    } on ApiException catch (err) {
      if (context.mounted) _snack(context, err.message);
    }
  }

  Future<void> _clearHistory(BuildContext context) async {
    final api = context.read<ApiClient>();
    final chat = context.read<ChatState>();
    final ok = await showConfirmDialog(
      context,
      title: 'Borrar historial',
      message: 'Se eliminarán TODAS tus conversaciones. '
          'Esta acción no se puede deshacer.',
      confirmLabel: 'Borrar todo',
      danger: true,
    );
    if (!ok) return;
    try {
      await api.delete('/api/account/conversations');
      chat.newChat();
      if (context.mounted) _snack(context, 'Historial eliminado.');
    } on ApiException catch (err) {
      if (context.mounted) _snack(context, err.message);
    }
  }

  Future<void> _deleteAccount(BuildContext context) async {
    final api = context.read<ApiClient>();
    final auth = context.read<AuthState>();
    final password = await showPromptDialog(
      context,
      title: 'Eliminar cuenta',
      message: 'Se borrarán tu cuenta, historial y claves de forma definitiva. '
          'Escribe tu contraseña para confirmar. (Si entraste con Google/Apple, '
          'primero crea una contraseña con "¿Olvidaste tu contraseña?").',
      placeholder: 'Contraseña',
      secure: true,
      confirmLabel: 'Eliminar',
      danger: true,
    );
    if (password == null || password.isEmpty) return;
    try {
      await api.delete('/api/account', body: {'password': password});
      await auth.logout(withNotice: 'Tu cuenta fue eliminada.');
    } on ApiException catch (err) {
      if (context.mounted) _snack(context, err.message);
    }
  }

  Future<void> _logout(BuildContext context) async {
    final auth = context.read<AuthState>();
    final ok = await showConfirmDialog(
      context,
      title: 'Cerrar sesión',
      message: '¿Salir de tu cuenta en este dispositivo?',
      confirmLabel: 'Cerrar sesión',
      danger: true,
    );
    if (ok) await auth.logout();
  }

  @override
  Widget build(BuildContext context) {
    final c = LixColors.of(context);
    final auth = context.watch<AuthState>();
    final prefs = context.watch<PrefsState>();
    final api = context.read<ApiClient>();
    final user = auth.user ?? {};
    final settings = (user['settings'] is Map)
        ? Map<String, dynamic>.from(user['settings'] as Map)
        : <String, dynamic>{};

    final fullName = [user['first_name'], user['last_name']]
        .whereType<String>()
        .where((s) => s.isNotEmpty)
        .join(' ');
    final email = (user['email'] as String?) ?? '';
    final initialsSource = fullName.isNotEmpty ? fullName : email;
    final initials = initialsSource.isNotEmpty
        ? initialsSource
            .split(RegExp(r'\s+'))
            .where((s) => s.isNotEmpty)
            .take(2)
            .map((s) => s[0].toUpperCase())
            .join()
        : '?';

    return Scaffold(
      backgroundColor: c.bg,
      body: SafeArea(
        child: ListView(
          padding: const EdgeInsets.fromLTRB(18, 10, 18, 40),
          children: [
            Text(
              'Cuenta',
              style: TextStyle(
                color: c.ink,
                fontFamily: kFontUi,
                fontWeight: FontWeight.w600,
                fontSize: 22,
              ),
            ),
            const SizedBox(height: 16),

            // Perfil
            _card(c, background: c.bgSecondary, children: [
              Row(
                children: [
                  CircleAvatar(
                    radius: 26,
                    backgroundColor: c.primary,
                    child: Text(
                      initials,
                      style: TextStyle(
                        color: c.onPrimary,
                        fontFamily: kFontUi,
                        fontWeight: FontWeight.w600,
                        fontSize: 18,
                      ),
                    ),
                  ),
                  const SizedBox(width: 14),
                  Expanded(
                    child: Column(
                      crossAxisAlignment: CrossAxisAlignment.start,
                      children: [
                        Text(
                          fullName.isNotEmpty
                              ? fullName
                              : ((user['username'] as String?) ?? '—'),
                          style: TextStyle(
                            color: c.ink,
                            fontFamily: kFontUi,
                            fontWeight: FontWeight.w600,
                            fontSize: 16,
                          ),
                        ),
                        Text(
                          email,
                          style: TextStyle(
                              color: c.inkSoft, fontFamily: kFontUi, fontSize: 13),
                        ),
                        if (user['plan_name'] is String) ...[
                          const SizedBox(height: 4),
                          Container(
                            padding: const EdgeInsets.symmetric(
                                horizontal: 10, vertical: 3),
                            decoration: BoxDecoration(
                              color: c.accent,
                              borderRadius: BorderRadius.circular(kRadiusPill),
                            ),
                            child: Text(
                              user['plan_name'] as String,
                              style: TextStyle(
                                color: c.onAccent,
                                fontFamily: kFontUi,
                                fontWeight: FontWeight.w500,
                                fontSize: 11,
                              ),
                            ),
                          ),
                        ],
                      ],
                    ),
                  ),
                  IconButton(
                    onPressed: () => _editName(context),
                    style: IconButton.styleFrom(backgroundColor: c.bg),
                    icon: Icon(Icons.edit, size: 16, color: c.ink),
                  ),
                ],
              ),
              if (user.isNotEmpty && user['email_verified'] != true) ...[
                const SizedBox(height: 10),
                InkWell(
                  onTap: () => _resendVerification(context),
                  borderRadius: BorderRadius.circular(12),
                  child: Container(
                    padding: const EdgeInsets.all(10),
                    decoration: BoxDecoration(
                      color: c.accentSoft,
                      borderRadius: BorderRadius.circular(12),
                    ),
                    child: Row(
                      children: [
                        Icon(Icons.mark_email_unread_outlined,
                            size: 15, color: c.accentDeep),
                        const SizedBox(width: 8),
                        Expanded(
                          child: Text(
                            'Correo sin verificar — toca para reenviar el enlace',
                            style: TextStyle(
                              color: c.accentDeep,
                              fontFamily: kFontUi,
                              fontWeight: FontWeight.w500,
                              fontSize: 12,
                            ),
                          ),
                        ),
                      ],
                    ),
                  ),
                ),
              ],
            ]),
            const SizedBox(height: 16),

            // Apariencia
            _card(c, children: [
              _section(c, 'Apariencia'),
              const SizedBox(height: 8),
              Row(
                children: [
                  _themeChip(c, prefs, 'Auto', ThemeMode.system),
                  const SizedBox(width: 8),
                  _themeChip(c, prefs, 'Claro', ThemeMode.light),
                  const SizedBox(width: 8),
                  _themeChip(c, prefs, 'Oscuro', ThemeMode.dark),
                ],
              ),
            ]),
            const SizedBox(height: 16),

            // Privacidad
            _card(c, children: [
              _section(c, 'Privacidad'),
              _switchRow(
                context,
                c,
                icon: Icons.save_outlined,
                label: 'Guardar historial',
                value: settings['save_history'] != false,
                onChanged: (v) => _toggleSetting(context, 'save_history', v),
              ),
              _switchRow(
                context,
                c,
                icon: Icons.insights_outlined,
                label: 'Métricas anónimas',
                value: settings['anonymous_usage'] != false,
                onChanged: (v) => _toggleSetting(context, 'anonymous_usage', v),
              ),
            ]),
            const SizedBox(height: 16),

            // Recursos y seguridad
            _card(c, children: [
              _actionRow(c,
                  icon: Icons.menu_book_outlined,
                  label: 'Documentación',
                  onTap: () => launchUrl(Uri.parse('${api.base}/docs'),
                      mode: LaunchMode.externalApplication)),
              Divider(height: 1, color: c.borderSoft),
              _actionRow(c,
                  icon: Icons.key_outlined,
                  label: 'Regenerar API key',
                  onTap: () => _regenerateKey(context)),
              Divider(height: 1, color: c.borderSoft),
              _actionRow(c,
                  icon: Icons.delete_outline,
                  label: 'Borrar historial de conversaciones',
                  onTap: () => _clearHistory(context)),
            ]),
            const SizedBox(height: 16),

            // Sesión
            _card(c, children: [
              _actionRow(c,
                  icon: Icons.logout,
                  label: 'Cerrar sesión',
                  onTap: () => _logout(context)),
              Divider(height: 1, color: c.borderSoft),
              _actionRow(c,
                  icon: Icons.warning_amber_outlined,
                  label: 'Eliminar cuenta',
                  danger: true,
                  onTap: () => _deleteAccount(context)),
            ]),
            const SizedBox(height: 20),

            Center(
              child: Text(
                'Lixbon móvil · v0.1.0',
                style:
                    TextStyle(color: c.inkMuted, fontFamily: kFontUi, fontSize: 11),
              ),
            ),
          ],
        ),
      ),
    );
  }

  Widget _card(LixColors c, {Color? background, required List<Widget> children}) {
    return Container(
      padding: const EdgeInsets.all(14),
      decoration: BoxDecoration(
        color: background ?? c.bg,
        borderRadius: BorderRadius.circular(kRadiusBox),
        border: Border.all(color: c.borderSoft),
      ),
      child: Column(crossAxisAlignment: CrossAxisAlignment.start, children: children),
    );
  }

  Widget _section(LixColors c, String text) => Padding(
        padding: const EdgeInsets.only(left: 4, bottom: 2),
        child: Text(
          text.toUpperCase(),
          style: TextStyle(
            color: c.inkSoft,
            fontFamily: kFontUi,
            fontWeight: FontWeight.w600,
            fontSize: 12,
            letterSpacing: 0.6,
          ),
        ),
      );

  Widget _switchRow(
    BuildContext context,
    LixColors c, {
    required IconData icon,
    required String label,
    required bool value,
    required ValueChanged<bool> onChanged,
  }) {
    return Padding(
      padding: const EdgeInsets.symmetric(vertical: 2, horizontal: 4),
      child: Row(
        children: [
          Icon(icon, size: 19, color: c.inkSoft),
          const SizedBox(width: 12),
          Expanded(
            child: Text(
              label,
              style: TextStyle(
                color: c.ink,
                fontFamily: kFontUi,
                fontWeight: FontWeight.w500,
                fontSize: 14,
              ),
            ),
          ),
          Switch(
            value: value,
            onChanged: onChanged,
            activeTrackColor: c.accentDeep,
            inactiveTrackColor: c.borderSoft,
          ),
        ],
      ),
    );
  }

  Widget _actionRow(
    LixColors c, {
    required IconData icon,
    required String label,
    required VoidCallback onTap,
    bool danger = false,
  }) {
    final color = danger ? c.danger : c.ink;
    return InkWell(
      onTap: onTap,
      child: Padding(
        padding: const EdgeInsets.symmetric(vertical: 13, horizontal: 4),
        child: Row(
          children: [
            Icon(icon, size: 19, color: danger ? c.danger : c.inkSoft),
            const SizedBox(width: 12),
            Expanded(
              child: Text(
                label,
                style: TextStyle(
                  color: color,
                  fontFamily: kFontUi,
                  fontWeight: FontWeight.w500,
                  fontSize: 14,
                ),
              ),
            ),
            Icon(Icons.chevron_right, size: 16, color: c.inkMuted),
          ],
        ),
      ),
    );
  }

  Widget _themeChip(LixColors c, PrefsState prefs, String label, ThemeMode mode) {
    final active = prefs.themeMode == mode;
    return InkWell(
      borderRadius: BorderRadius.circular(kRadiusPill),
      onTap: () => prefs.setThemeMode(mode),
      child: Container(
        padding: const EdgeInsets.symmetric(horizontal: 18, vertical: 9),
        decoration: BoxDecoration(
          color: active ? c.primary : c.bgSecondary,
          borderRadius: BorderRadius.circular(kRadiusPill),
        ),
        child: Text(
          label,
          style: TextStyle(
            color: active ? c.onPrimary : c.inkSoft,
            fontFamily: kFontUi,
            fontWeight: FontWeight.w500,
            fontSize: 13,
          ),
        ),
      ),
    );
  }
}

File: apps/mobile/src/screens/AccountScreen.js
// AccountScreen.js — todo lo del usuario: perfil, verificación de correo,
// tema, privacidad, API key, documentación (abre la web), borrado de
// historial, cierre de sesión y eliminación de cuenta. Espejo de "Mi cuenta"
// de la web (account.css): página crema con tarjetas blancas y títulos de
// tarjeta (18px semibold), avatar de tinta con inicial y plan-pill de acento.
import Constants from 'expo-constants';
import React from 'react';
import { Linking, Pressable, ScrollView, Switch, Text, View } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';

import Icon from '../components/Icon';
import { useDialogs } from '../components/dialogs';
import { Card, CardTitle, IconButton, PlanPill, StackHeader, useColors } from '../components/ui';
import { ApiException } from '../api';
import { useApi, useAuth, useChat, usePrefs } from '../state';
import { FONTS, RADIUS_PILL } from '../theme';

export default function AccountScreen({ onBack }) {
  const c = useColors();
  const api = useApi();
  const auth = useAuth();
  const chat = useChat();
  const prefs = usePrefs();
  const { prompt, confirm, toast } = useDialogs();

  const user = auth.user || {};
  const settings = user.settings && typeof user.settings === 'object' ? user.settings : {};

  const fullName = [user.first_name, user.last_name]
    .filter((s) => typeof s === 'string' && s)
    .join(' ');
  const email = typeof user.email === 'string' ? user.email : '';
  const initialsSource = fullName || email;
  const initials = initialsSource
    ? initialsSource
        .split(/\s+/)
        .filter(Boolean)
        .slice(0, 2)
        .map((s) => s[0].toUpperCase())
        .join('')
    : '?';

  const fail = (err) =>
    toast(err instanceof ApiException ? err.message : 'Sin conexión con el servidor');

  const editName = async () => {
    const value = await prompt({
      title: 'Tu nombre',
      message: 'Nombre y apellido separados por un espacio.',
      placeholder: 'Nombre Apellido',
      initialValue: fullName,
      confirmLabel: 'Guardar',
    });
    const trimmed = (value || '').trim();
    if (!trimmed) return;
    const parts = trimmed.split(/\s+/);
    const firstName = parts[0];
    const lastName = parts.length > 1 ? parts.slice(1).join(' ') : user.last_name || '';
    try {
      await api.patch('/api/account/profile', { first_name: firstName, last_name: lastName });
      await auth.refreshMe();
    } catch (err) {
      fail(err);
    }
  };

  const toggleSetting = async (key, value) => {
    try {
      await api.patch('/api/account/settings', { [key]: value });
      await auth.refreshMe();
    } catch (err) {
      fail(err);
    }
  };

  const resendVerification = async () => {
    try {
      const res = await api.post('/api/auth/resend-verification');
      toast(String(res?.message || 'Correo enviado'));
    } catch (err) {
      fail(err);
    }
  };

  const regenerateKey = async () => {
    const ok = await confirm({
      title: 'Regenerar API key',
      message:
        'Tu key actual dejará de funcionar en todos los dispositivos ' +
        '(esta app adoptará la nueva automáticamente).',
      confirmLabel: 'Regenerar',
      danger: true,
    });
    if (!ok) return;
    try {
      const res = await api.post('/api/auth/api-key/regenerate');
      if (typeof res?.newApiKey === 'string' && res.newApiKey) {
        await auth.adoptApiKey(res.newApiKey);
      }
      toast('API key regenerada. Esta app ya usa la nueva.');
    } catch (err) {
      fail(err);
    }
  };

  const clearHistory = async () => {
    const ok = await confirm({
      title: 'Borrar historial',
      message: 'Se eliminarán TODAS tus conversaciones. Esta acción no se puede deshacer.',
      confirmLabel: 'Borrar todo',
      danger: true,
    });
    if (!ok) return;
    try {
      await api.delete('/api/account/conversations');
      chat.newChat();
      toast('Historial eliminado.');
    } catch (err) {
      fail(err);
    }
  };

  const deleteAccount = async () => {
    const password = await prompt({
      title: 'Eliminar cuenta',
      message:
        'Se borrarán tu cuenta, historial y claves de forma definitiva. ' +
        'Escribe tu contraseña para confirmar. (Si entraste con Google/Apple, ' +
        'primero crea una contraseña con "¿Olvidaste tu contraseña?").',
      placeholder: 'Contraseña',
      secure: true,
      confirmLabel: 'Eliminar',
      danger: true,
    });
    if (!password) return;
    try {
      await api.delete('/api/account', { password });
      await auth.logout('Tu cuenta fue eliminada.');
    } catch (err) {
      fail(err);
    }
  };

  const logout = async () => {
    const ok = await confirm({
      title: 'Cerrar sesión',
      message: '¿Salir de tu cuenta en este dispositivo?',
      confirmLabel: 'Cerrar sesión',
      danger: true,
    });
    if (ok) await auth.logout();
  };

  const version = Constants.expoConfig?.version || '';

  return (
    <SafeAreaView edges={['top']} style={{ flex: 1, backgroundColor: c.bgSecondary }}>
      <StackHeader title="Cuenta" onBack={onBack} />
      <ScrollView contentContainerStyle={{ paddingBottom: 40, paddingTop: 6 }}>
        <View style={{ paddingHorizontal: 16, gap: 14 }}>
          {/* Perfil */}
          <Card>
            <View style={{ flexDirection: 'row', alignItems: 'center' }}>
              <View
                style={{
                  width: 52,
                  height: 52,
                  borderRadius: 26,
                  backgroundColor: c.primary,
                  alignItems: 'center',
                  justifyContent: 'center',
                }}
              >
                <Text style={{ fontFamily: FONTS.uiSemiBold, fontSize: 18, color: c.onPrimary }}>
                  {initials}
                </Text>
              </View>
              <View style={{ flex: 1, marginLeft: 14, gap: 2 }}>
                <Text style={{ fontFamily: FONTS.uiSemiBold, fontSize: 16, color: c.ink }}>
                  {fullName || user.username || '—'}
                </Text>
                <Text style={{ fontFamily: FONTS.ui, fontSize: 13.5, color: c.inkSoft }}>
                  {email}
                </Text>
                {typeof user.plan_name === 'string' && (
                  <View style={{ marginTop: 4 }}>
                    <PlanPill>{user.plan_name}</PlanPill>
                  </View>
                )}
              </View>
              <IconButton onPress={editName}>
                <Icon name="pencil" size={16} color={c.ink} />
              </IconButton>
            </View>

            {Object.keys(user).length > 0 && user.email_verified !== true && (
              <Pressable
                onPress={resendVerification}
                style={({ pressed }) => ({
                  marginTop: 12,
                  flexDirection: 'row',
                  alignItems: 'center',
                  gap: 8,
                  paddingHorizontal: 12,
                  paddingVertical: 10,
                  borderRadius: 12,
                  backgroundColor: c.accentSoft,
                  opacity: pressed ? 0.72 : 1,
                })}
              >
                <Icon name="mail" size={15} color={c.accentDeep} />
                <Text
                  style={{ flex: 1, fontFamily: FONTS.uiMedium, fontSize: 13, color: c.accentDeep }}
                >
                  Correo sin verificar — toca para reenviar el enlace
                </Text>
              </Pressable>
            )}
          </Card>

          {/* Apariencia */}
          <Card>
            <CardTitle style={{ marginBottom: 12 }}>Apariencia</CardTitle>
            <View style={{ flexDirection: 'row', gap: 8 }}>
              {[
                ['Auto', 'system'],
                ['Claro', 'light'],
                ['Oscuro', 'dark'],
              ].map(([label, mode]) => {
                const active = prefs.themeMode === mode;
                return (
                  <Pressable
                    key={mode}
                    onPress={() => prefs.setThemeMode(mode)}
                    style={({ pressed }) => ({
                      paddingHorizontal: 18,
                      paddingVertical: 9,
                      borderRadius: RADIUS_PILL,
                      backgroundColor: active ? c.primary : 'transparent',
                      borderWidth: 1,
                      borderColor: active ? c.primary : c.borderSoft,
                      opacity: pressed && !active ? 0.72 : 1,
                    })}
                  >
                    <Text
                      style={{
                        fontFamily: FONTS.uiMedium,
                        fontSize: 13,
                        color: active ? c.onPrimary : c.ink,
                      }}
                    >
                      {label}
                    </Text>
                  </Pressable>
                );
              })}
            </View>
          </Card>

          {/* Privacidad */}
          <Card>
            <CardTitle style={{ marginBottom: 6 }}>Privacidad</CardTitle>
            <SwitchRow
              icon="save"
              label="Guardar historial"
              value={settings.save_history !== false}
              onChange={(v) => toggleSetting('save_history', v)}
            />
            <SwitchRow
              icon="activity"
              label="Métricas anónimas"
              value={settings.anonymous_usage !== false}
              onChange={(v) => toggleSetting('anonymous_usage', v)}
            />
          </Card>

          {/* Recursos y seguridad */}
          <Card style={{ paddingVertical: 8 }}>
            <ActionRow
              icon="book"
              label="Documentación"
              onPress={() => Linking.openURL(`${api.base}/docs`)}
            />
            <Separator />
            <ActionRow icon="key" label="Regenerar API key" onPress={regenerateKey} />
            <Separator />
            <ActionRow
              icon="trash"
              label="Borrar historial de conversaciones"
              onPress={clearHistory}
            />
          </Card>

          {/* Sesión */}
          <Card style={{ paddingVertical: 8 }}>
            <ActionRow icon="logout" label="Cerrar sesión" onPress={logout} />
            <Separator />
            <ActionRow icon="warning" label="Eliminar cuenta" danger onPress={deleteAccount} />
          </Card>

          <Text
            style={{
              textAlign: 'center',
              marginTop: 6,
              fontFamily: FONTS.ui,
              fontSize: 12,
              color: c.inkMuted,
            }}
          >
            Lixbon móvil · v{version}
          </Text>
        </View>
      </ScrollView>
    </SafeAreaView>
  );
}

function Separator() {
  const c = useColors();
  return <View style={{ height: 1, backgroundColor: c.borderSoft }} />;
}

function SwitchRow({ icon, label, value, onChange }) {
  const c = useColors();
  return (
    <View
      style={{
        flexDirection: 'row',
        alignItems: 'center',
        gap: 12,
        paddingVertical: 4,
      }}
    >
      <Icon name={icon} size={19} color={c.inkSoft} />
      <Text style={{ flex: 1, fontFamily: FONTS.ui, fontSize: 14.5, color: c.ink }}>{label}</Text>
      <Switch
        value={value}
        onValueChange={onChange}
        trackColor={{ true: c.accentDeep, false: c.track }}
        thumbColor="#FFFFFF"
      />
    </View>
  );
}

function ActionRow({ icon, label, onPress, danger = false }) {
  const c = useColors();
  return (
    <Pressable
      onPress={onPress}
      style={({ pressed }) => ({
        flexDirection: 'row',
        alignItems: 'center',
        gap: 12,
        paddingVertical: 14,
        paddingHorizontal: 4,
        marginHorizontal: -4,
        borderRadius: 12,
        backgroundColor: pressed ? c.pressed : 'transparent',
      })}
    >
      <Icon name={icon} size={19} color={danger ? c.danger : c.inkSoft} />
      <Text
        style={{
          flex: 1,
          fontFamily: FONTS.ui,
          fontSize: 14.5,
          color: danger ? c.danger : c.ink,
        }}
      >
        {label}
      </Text>
      <Icon name="chevron-right" size={16} color={c.inkMuted} />
    </Pressable>
  );
}
